types/underscore/underscore-tests.ts

```diff
@@ -2976,6 +2977,102 @@ undefinedIdentityIterateeResult; // $ExpectType StringRecord
     extractChainTypes(_.chain(mixedTypeRecord).findKey()); // $ExpectType ChainType<"a" | "b" | "c" | undefined, string>
 }
 
+// pick
+{
+    // individual key parameters - record
+    _.pick(mixedTypeRecord, 'a', 'b'); // $ExpectType Pick<MixedTypeRecord, "a" | "b">
+    _(mixedTypeRecord).pick('a', 'b'); // $ExpectType Pick<MixedTypeRecord, "a" | "b">
+    extractChainTypes(_.chain(mixedTypeRecord).pick('a', 'b')); // $ExpectType ChainType<Pick<MixedTypeRecord, "a" | "b">, number | StringRecord>
+
+    // individual key parameters - any
+    _.pick(anyValue, 'a', 'b'); // $ExpectType Pick<any, "a" | "b">
+    _(anyValue).pick('a', 'b'); // $ExpectType Pick<any, "a" | "b">
+    extractChainTypes(_.chain(anyValue).pick('a', 'b')); // $ExpectType ChainType<Pick<any, "a" | "b">, any>
+
+    // key array - record
+    _.pick(mixedTypeRecord, ['a', 'b']); // $ExpectType Pick<MixedTypeRecord, "a" | "b">
+    _(mixedTypeRecord).pick(['a', 'b']); // $ExpectType Pick<MixedTypeRecord, "a" | "b">
+    extractChainTypes(_.chain(mixedTypeRecord).pick(['a', 'b'])); // $ExpectType ChainType<Pick<MixedTypeRecord, "a" | "b">, number | StringRecord>
+
+    // key array - any
+    _.pick(anyValue, ['a', 'b']); // $ExpectType Pick<any, "a" | "b">
+    _(anyValue).pick(['a', 'b']); // $ExpectType Pick<any, string>
+    extractChainTypes(_.chain(anyValue).pick(['a', 'b'])); // $ExpectType ChainType<Pick<any, string>, any>
```

Comment:
I'm a bit confused about the inconsistency here, and unfortunately I haven't found a reason for it or a way to make it not happen. I don't feel like it's a big enough deal to try and fight it further at this point, but I'd be happy to take another look or switch to a solution similar to `_Omit` where the result for `any` is always `any`.

`Pick<any, "a" | "b">` seems to be equivalent to type `{ a: any; b: any; }`.

Reply:
I agree it's mysterious. I don't have an opinion on what to do here.

Reply:
Good news! By moving arrays to the first overload the mystery mysteriously goes away!